### DISCUSSION TITLE
Some remarks on recent changes

### DISCUSSION THREAD

**Discussion by charcoalbe** [plule/theremotion#60] [General]
First of all, I'm not sure if this is the good place to put remarks as well; i'm not that advanced in github ways of working but I wanted to give some feedback after using theremotion since its release. I do not mean this as a way of criticism at all, I'm very grateful for this software and am finally using the leap motion controller that has been gathering dust for the last few years.

Things have changed a lot since the first version. I get that the movement now resembles a real Theremin more but after two weeks of playing with vertical pitch, it's quite a bit change that was implemented in 1.2. 

1. fullscreen does not seem to work on my device (Surface Pro 7+); it just opens up in the left top corner of my screen
2. playing standing up I seem to hit the chords very quickly, playing sitting down has other difficulties (especially about reaching full range of sounds)
3. would be neat if disabling the chords view also disabled the chords when playing, now it only disables the visual
4. I'm not sure pinching my right hand is noticed quickly enough to discover the "snap to scale". Having some difficulties reaching full pitch range there as well. 
5. When not pinching right hand, I wonder if the note display should somehow indicate this (a bit like the diagonal would vary in previous versions). 

Thanks a lot for all your hard work!

**Comment by plule** [comment]
Hey there @charcoalbe 

> First of all, I'm not sure if this is the good place to put remarks as well; i'm not that advanced in github ways of working but I wanted to give some feedback after using theremotion since its release. I do not mean this as a way of criticism at all, I'm very grateful for this software and am finally using the leap motion controller that has been gathering dust for the last few years.

This is very welcomed! Theremotion being a week-end project, having people taking the time to learn how to use it, and giving feedback is the most motivating thing I can think of. I've created the "Discussion" section of github and moved your post here, to get a discussion place less frightening than "issues" :)

> Things have changed a lot since the first version. I get that the movement now resembles a real Theremin more but after two weeks of playing with vertical pitch, it's quite a bit change that was implemented in 1.2.

Yeah, I can imagine... If it make you feel better, know that every single interaction was reworked many times, so I had to relearn it soooo many time, but this last change was among the most impacting. In the end I'm very happy with it, the thing that convinced me is that not only it keeps the Theremin design language, but in a way it expands it: The vertical axis is a sort of volume for both hands (volume, and additional chords). The horizontal axis becomes the shaping of the sound, with the pitch and filter acting together.

With that being said, I'm adding a word of warning in the doc, I kind of rushed a first "presentable" version as I did not expect people to actually pick it up and seriously try it, but in a way I'm still constantly thinking about the general design, so it's not impossible to have other major changes in the future. I reaaaally need to make a tutorial video here.

> fullscreen does not seem to work on my device (Surface Pro 7+); it just opens up in the left top corner of my screen

It will be better on next release. Up until now I hardcoded all the dimensions to fit well on the small 800x600 touchscreen I'm using, but I've started reworking it to make the UI actually scale.

> playing standing up I seem to hit the chords very quickly, playing sitting down has other difficulties (especially about reaching full range of sounds)

I can see why. I've been pushing back a bit on adding measurement settings, but maybe it's time to add some. I'm a bit afraid to make it less reliable in term of muscle memory by making it configurable, but it's really dependent on both the environment and player's morphology. I'll try to add some settings for a further release here.

> would be neat if disabling the chords view also disabled the chords when playing, now it only disables the visual

Well, I actually just discovered that it's possible to disable the "chord" thing in the UI thanks to your comment. Here I'm kind of abusing the legend of the plots to use it as a title, and it happen that you can disable it. That being said, I like the idea of being able to disable the auto-chord in a preset, especially given that the algorithm does not play well will all the scales.

> I'm not sure pinching my right hand is noticed quickly enough to discover the "snap to scale". Having some difficulties reaching full pitch range there as well.

Can you elaborate here? I'm super curious but I can't understand what you mean by "noticed quickly enough to discover the snap to scale".

> When not pinching right hand, I wonder if the note display should somehow indicate this (a bit like the diagonal would vary in previous versions).

Yeees I feel like the new view is an improvement on nearly all front, except this one. The problem is that I can't find any idea on how it could be represented.

**Reply by charcoalbe** [reply]
### general ramble

> > Things have changed a lot since the first version. I get that the movement now resembles a real Theremin more but after two weeks of playing with vertical pitch, it's quite a bit change that was implemented in 1.2.
> 
> Yeah, I can imagine... If it make you feel better, know that every single interaction was reworked many times, so I had to relearn it soooo many time, but this last change was among the most impacting. In the end I'm very happy with it, the thing that convinced me is that not only it keeps the Theremin design language, but in a way it expands it: The vertical axis is a sort of volume for both hands (volume, and additional chords). The horizontal axis becomes the shaping of the sound, with the pitch and filter acting together.
> 

I can only imagine the work that goes in to a project like this. I discovered you on reddit (nickname "haaid" overthere) and have been trying to figure out how to play better. Not anywhere near achieving your demo video though. The adding of the guitar is nice; reminds me of some other instrument (Hurdy-Gurdy / Vielle a roue) that has a "trumpet" string to give a sort of rythm to whatever you play. 

I've tried some variations such as switching off the lead and only "picking" certain tunes; it's a playfield for me as well obviously :)

### on pinching the right hand

> > I'm not sure pinching my right hand is noticed quickly enough to discover the "snap to scale". Having some difficulties reaching full pitch range there as well.
> 
> Can you elaborate here? I'm super curious but I can't understand what you mean by "noticed quickly enough to discover the snap to scale".

I find the right hand in version 1.2 to be a bit unreliable. It notices proximity before it notices pinched or not I find (but this might very well be because of my position when playing, which i generally do just seated at a table). 

### on display

> > When not pinching right hand, I wonder if the note display should somehow indicate this (a bit like the diagonal would vary in previous versions).
> 
> Yeees I feel like the new view is an improvement on nearly all front, except this one. The problem is that I can't find any idea on how it could be represented.

The look and feel of the circles is a lot better than the diagonal line; even when it becomes visually unclear whether or not you are working with fixed notes (pinched) or fluid (unpinched). It may be as easy to make the main dot change color and have it red when pinched and blue when not (or any other variety of color). 


### on drones

I don't seem to notice anything when changing drone volume. It might be that my simple soundcard doesn't make anything of it but I don't have the idea i actually hear a drone (nor am I aware where I should hear it). As mentioned before, I do refer to other instruments that have "drones" such as the afformentioned hurdy-gurdy but also bagpipes and a few other traditional folk instruments; these are generally base-notes that you keep on hearing no matter what you play (and therefore most useful if you play in a fixed scale. . If you want to enable something like that it might be best to also allow for changing the scales manually. Now you can save them as a part of a total preset but not as a scale itself.

**Reply by plule** [reply]
>The adding of the guitar is nice; reminds me of some other instrument (Hurdy-Gurdy / Vielle a roue) that has a "trumpet" string to give a sort of rythm to whatever you play.

Ooooh that's so funny. The Theremotion is quite inspired by the Hurdy Gurdy in the sense that it plays on a set scale with a drone, but I've never considered the guitar to be part of the inspiration. It gives me ideas on how to modulate the drone voice, like maybe "grabbing" instead of "pinching" the volume hand would make a similar effect to the trumpet string.

> I find the right hand in version 1.2 to be a bit unreliable. It notices proximity before it notices pinched or not I find (but this might very well be because of my position when playing, which i generally do just seated at a table).

Of all your remarks, this one worries me so much, because there is no expected change here in 1.2, but internally I've changed a lot on how the pieces of the app communicates...
 
> The look and feel of the circles is a lot better than the diagonal line; even when it becomes visually unclear whether or not you are working with fixed notes (pinched) or fluid (unpinched). It may be as easy to make the main dot change color and have it red when pinched and blue when not (or any other variety of color).

I had in mind complicated stuff like representing thicker circle representing the autoscaled area, but color change is super simple and easy to implement, that's a nice idea.

> I don't seem to notice anything when changing drone volume. It might be that my simple soundcard doesn't make anything of it but I don't have the idea i actually hear a drone (nor am I aware where I should hear it). As mentioned before, I do refer to other instruments that have "drones" such as the afformentioned hurdy-gurdy but also bagpipes and a few other traditional folk instruments; these are generally base-notes that you keep on hearing no matter what you play (and therefore most useful if you play in a fixed scale. . If you want to enable something like that it might be best to also allow for changing the scales manually. Now you can save them as a part of a total preset but not as a scale itself.

Well well, this is totally undocumented and unexplained in the UI, but the keyboard is interactive and depends on the tab:

- In the "Play" tab, you can use it to activate up to 4 drone notes. I think you did not have any drone note playing, so the volume was not changing anything
- In the "Root note" tab it can be used to select the root note
- In the "Scale" tab it can be used to select a manual scale instead of one of the prepared one. By the way if you have relevant scales that should be included here to be in the spirit of hurdy-gurdies and bagpipes, I'm interested, I'm lacking on the knowledge of the folk music scales
